src/Writer/ResourceWriter.php

```diff
@@ -174,38 +164,14 @@ public function writeResources() // _putresources
 		}
 
 		/* -- BACKGROUNDS -- */
-		if (($this->mpdf->gradients !== null && (count($this->mpdf->gradients) > 0)) || ($this->mpdf->enableImports && count($this->mpdf->tpls))) { // mPDF 5.7.3
+		if (($this->mpdf->gradients !== null && (count($this->mpdf->gradients) > 0))) { // mPDF 5.7.3
 
 			$this->writer->write('/Shading <<');
 
 			foreach ($this->mpdf->gradients as $id => $grad) {
 				$this->writer->write('/Sh' . $id . ' ' . $grad['id'] . ' 0 R');
 			}
 
-			// mPDF 5.7.3
-			// If a shading dictionary is in an object (tpl) imported from another PDF, it needs to be included
-			// in the document resources, as well as the object resources
-			// Otherwise get an error in some PDF viewers
-			if ($this->mpdf->enableImports && count($this->mpdf->tpls)) {
-
-				foreach ($this->mpdf->tpls as $tplidx => $tpl) {
-
-					if (isset($tpl['resources'])) {
-
-						$this->mpdf->current_parser = $tpl['parser'];
-
-						foreach ($tpl['resources'][1] as $k => $v) {
-							if ($k === '/Shading') {
-								foreach ($v[1] as $k2 => $v2) {
-									$this->writer->write($k2 . ' ', false);
-									$this->mpdf->pdf_write_value($v2);
-								}
-							}
-						}
-					}
-				}
-			}
-
```

Comment:
We never received any report for FPDI regarding such behavior/issue. Maybe it was an issue in a viewer application but from the technical view this is simply not required.